### PR TITLE
boost1.83, add new version

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
Filed an issue upstream for the missing boost_locale library: https://github.com/boostorg/boost/issues/799

In the beginning of the build process I see it being used?
```
- locale                   : building
```

File: dev-libs/boost/boost183-1.83.0.recipe
Comment (jmairboeck) [inline dev-libs/boost/boost183-1.83.0.recipe:1]: Just a suggestion: maybe rename the recipe to `boost1.83-1.83.0.recipe`, i.e. include the `.` in the package name, similar to python?

**Comment by Begasus** [comment]
I folowed the naming scheme as has been done for all other boost recipes, I could switch if it would be advisable though.

**Comment by pulkomandy** [comment]
Yes, I think the name with a dot is better and should be used for new recipes.

**Comment by Begasus** [comment]
Recipe name has been changed. :)

**Comment by johnsonjh** [comment]
@Begasus 

I can't build this ... :(

```sh
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-13/release/inlining-on/threading-multi/visibility-hidden/directory.o                                                                             
libs/filesystem/src/directory.cpp: In function 'std::size_t boost::filesystem::detail::{anonymous}::get_path_max()':                                                                              
libs/filesystem/src/directory.cpp:205:21: error: invalid application of 'sizeof' to incomplete type 'char []'                                                                                     
  205 |     if ((max + 1) < sizeof(dirent().d_name))                                                                                                                                              
      |                     ^~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                               
libs/filesystem/src/directory.cpp:206:15: error: invalid application of 'sizeof' to incomplete type 'char []'                                                                                     
  206 |         max = sizeof(dirent().d_name) - 1;                                                                                                                                                
      |               ^~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                     
libs/filesystem/src/directory.cpp: In function 'boost::system::error_code boost::filesystem::detail::{anonymous}::dir_itr_create(boost::intrusive_ptr<boost::filesystem::detail::dir_itr_imp>&, const boost::filesystem::path&, unsigned int, boost::filesystem::detail::directory_iterator_params*, boost::filesystem::path&, boost::filesystem::file_status&, boost::filesystem::file_status&)':  
libs/filesystem/src/directory.cpp:368:44: error: invalid application of 'sizeof' to incomplete type 'char []'                                                                                     
  368 |             extra_size = (sizeof(dirent) - sizeof(dirent().d_name)) + path_max() + 1u; // + 1 for "\0"                                                                                    
      |                                            ^~~~~~~~~~~~~~~~~~~~~~~                                                                                                                        
                                                                                                                                                                                                  
    "g++"   -fvisibility-inlines-hidden -fPIC -m64 -O3 -Wno-inline -Wall -fvisibility=hidden  -DBOOST_ALL_NO_LIB=1 -DBOOST_ATOMIC_DYN_LINK=1 -DBOOST_FILESYSTEM_DYN_LINK=1 -DBOOST_FILESYSTEM_HAS_FDOPENDIR_NOFOLLOW -DBOOST_FILESYSTEM_HAS_INIT_PRIORITY -DBOOST_FILESYSTEM_HAS_STAT_ST_BLKSIZE -DBOOST_FILESYSTEM_HAS_STAT_ST_MTIM -DBOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF -DBOOST_FILESYSTEM_SOURCE -DNDEBUG   -I"." -I"libs/filesystem/src"  -c -o "bin.v2/libs/filesystem/build/gcc-13/release/inlining-on/threading-multi/visibility-hidden/directory.o" "libs/filesystem/src/directory.cpp"     

```

Am I doing it wrong?

**Comment by korli** [comment]
> Am I doing it wrong?

Looks a bit like https://github.com/haikuports/haikuports/blob/1a145f36ae170e7164226cf3e94d0b4c678daa0f/dev-libs/boost/patches/boost-1.69.0.patchset#L186

**Comment by Begasus** [comment]
> I can't build this ... :(

Is this on 32bit? I got no issues building it on 64bit, haven't checked 32bit (yet).

**Comment by Begasus** [comment]
@johnsonjh Builds OK here (with the PCH issue out of the way):
`grabbing boost1.83_x86-1.83.0-1-x86_gcc2.hpkg and moving it to /boot/home/haikuports/packages/boost1.83_x86-1.83.0-1-x86_gcc2.hpkg ...`

**Comment by Begasus** [comment]
Have been checking this with quite a number of recipes still using `devel:libboost_system >= 1.69.0`, so far all things worked out OK.
Build on 32bit was fine here too (with the latest changes included).

**Pull request merged**